compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -1579,7 +1579,7 @@ impl<'a> Parser<'a> {
         self.expect(&token::ModSep)?;
 
         let mut path = ast::Path { segments: ThinVec::new(), span: DUMMY_SP, tokens: None };
-        self.parse_path_segments(&mut path.segments, T::PATH_STYLE, None)?;
+        self.parse_path_segments(&mut path.segments, T::PATH_STYLE, None, None)?;
```

Comment:
I'm not super happy with the proliferation of `None`s this causes, but I also don't have much of a better proposal (short of having a `*_with_pat_location` family of methods).

Reply:
The new changes reduce the `None`s, but there are still some (seemingly unavoidable at the moment)